src/components/tabController/FadedScrollView.tsx

```diff
@@ -1,47 +1,80 @@
+import React, {useCallback, useImperativeHandle} from 'react';
 import {
-  ViewProps,
-  ScrollView,
+  ScrollView as RNScrollView,
   ScrollViewProps,
   NativeSyntheticEvent,
   NativeScrollEvent,
   LayoutChangeEvent
 } from 'react-native';
-import Fader from '../fader';
+import Fader, {FaderProps} from '../fader';
 import useScrollEnabler from '../../hooks/useScrollEnabler';
 import useScrollReached from '../../hooks/useScrollReached';
-import forwardRef, {ForwardRefInjectedProps} from '../../commons/forwardRef';
+import {forwardRef, ForwardRefInjectedProps} from '../../commons/new';
+import {ScrollView as GestureScrollView} from 'react-native-gesture-handler';
 
-export type FadedScrollViewProps = ViewProps &
-  ScrollViewProps & {
-    children?: React.ReactNode | React.ReactNode[];
-  };
+export type FadedScrollViewProps = ScrollViewProps & {
+  /**
+   * Show a fader at the start of the scroll
+   */
+  showStartFader?: boolean;
+  /**
+   * Additional props for the start fader
+   */
+  startFaderProps?: Omit<FaderProps, 'visible' | 'position'>;
+  /**
+   * Show a fader at the end of the scroll
+   */
+  showEndFader?: boolean;
+  /**
+   * Additional props for the end fader
+   */
+  endFaderProps?: Omit<FaderProps, 'visible' | 'position'>;
+  /**
+   * Use the react-native-gesture-handler version, useful when using react-native-reanimated
+   */
+  useGesture?: boolean;
+  children?: React.ReactNode | React.ReactNode[];
+};
 
 type Props = FadedScrollViewProps & ForwardRefInjectedProps;
-
-const FADER_SIZE = 76;
+interface Statics {
+  scrollTo(
+    y?: number | {x?: number | undefined; y?: number | undefined; animated?: boolean | undefined},
+    x?: number,
+    animated?: boolean
+  ): void;
+  isScrollEnabled: () => boolean;
+}
 
 const FadedScrollView = (props: Props) => {
   const {
     children,
     onScroll: propsOnScroll,
     onContentSizeChange: propsOnContentSizeChange,
     onLayout: propsOnLayout,
-    ...other
+    horizontal: propsHorizontal,
+    showStartFader,
+    startFaderProps,
+    showEndFader,
+    endFaderProps,
+    useGesture,
+    ...others
   } = props;
-  const ref = useRef<ScrollView>();
-  const {onContentSizeChange, onLayout, scrollEnabled} = useScrollEnabler({horizontal: true});
+  const ScrollView = useGesture ? GestureScrollView : RNScrollView;
+  const scrollViewRef = React.createRef<typeof ScrollView>();
+  const horizontal = propsHorizontal ?? false;
+  const {onContentSizeChange, onLayout, scrollEnabled} = useScrollEnabler({horizontal});
   const {
     onScroll: onScrollReached,
     isScrollAtStart,
     isScrollAtEnd
   } = useScrollReached({
-    horizontal: true,
+    horizontal,
     threshold: 50
   });
 
-  const showStart = scrollEnabled && !isScrollAtStart;
-  const showEnd = scrollEnabled && !isScrollAtEnd;
+  const showStart = (showStartFader && scrollEnabled && !isScrollAtStart) ?? false;
+  const showEnd = (showEndFader && scrollEnabled && !isScrollAtEnd) ?? false;
```

Comment:
Why do you need the `?? false` in these two lines?

Reply:
The default for the `Fader` is to be visible, I want it to not be visible unless it's needed.